File: swift/easy/1-two-sum.swift
func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
    // declare map to collect pairs as number -> its index in the array
    var map = [Int: Int]()

    // iterate over numbers in the array
    for i in 0..<nums.count {
        // get current number
        let num = nums[i]
        // calculate second number to have sum of target
        let diff = target - num
        // check if has already passed second number
        if let j = map[diff] {
            // if yes, return indices
            return [j, i]
        } else {
            // if no, save current number and its index
            map[num] = i
        }
    }

    // should never be here
    return [-1, -1]
}

let ans = twoSum([2,7,11,15], 9)
print(ans)

File: easy/1.two-sum.java
public int[] twoSum(int[] nums, int target) {
    Map<Integer, Integer> map = new HashMap<Integer, Integer>();
    for (int i=0; i < nums.length; i++) {
        int num = nums[i];
        int diff = target - num;

        if (map.containsKey(diff)) {
            return new int[] {map.get(diff), i};
        }
        map.put(num, i);
    }

    return new int[] {-1, -1};
}
